app/models/patient_summary.rb

```diff
@@ -1,7 +1,17 @@
 class PatientSummary < ActiveRecord::Base
   self.primary_key = :id
 
+  belongs_to :patient, foreign_key: :id
+  belongs_to :next_appointment, class_name: 'Appointment', foreign_key: :next_appointment_id
+
   def readonly?
     true
   end
+
+  def self.overdue
+    where(next_appointment_status: 'scheduled')
+      .where('next_appointment_scheduled_date < ?', Date.current)
+      .where('next_appointment_scheduled_date >= ?', 365.days.ago)
+      .where('next_appointment_remind_on IS NULL OR next_appointment_remind_on <= ?', Date.current)
+  end
```

Comment:
`.overdue` could be a scope.

Just a thought, if we named `next_appointment_scheduled_date` as `scheduled_date` and so on, this scope could be re-used across `Appointment` and `PatientSummary`. Might be worth doing to avoid duplication.

Reply:
I considered using a scope, but avoided it because of some silly multi-line lambda code style issues. On second thought, I think a scope is the way to go.

I'm hesitant to rename this column since the prefix is basically a namespace to avoid clashes with joined tables. Imagine a scenario where we include the latest scheduled and latest visited appointment in this table 🤔 I'll give some thought to avoiding this duplication somehow.

Reply:
@prabhanshuguptagit I think I've got it. I've reworked the `PatientSummary.overdue` scope to simply re-use the `Appointment.overdue` scope through a `#merge()`. Check it out in 63cea05

This avoids duplication and also doesn't require a bunch of column name juggling.

Reply:
Neat 👍